Home sekcja kliknij na mnie

anetagloc opened the pull request:
I created a ‘Click on me’ section with interaction and styles in tailwind. I added the icons and the photo necessary to make the section.

PiotrDrechsler commented on src/app/components/ClickOnMe/ClickOnMeTitle.tsx:
```diff
@@ -0,0 +1,11 @@
+      </div>
+      </div>
+    );
+  };
```
It would be nice if you type a component and add a text props to it and then add the import to the main component. Also try to make such changes to the other smaller components: Here is an example of how to do it:

```
import React from 'react';

interface ClickOnMeTitleProps {
  text: string; // Typ dla prop 'text'
}

export const ClickOnMeTitle: React.FC<ClickOnMeTitleProps> = ({ text }) => {
  return (
    <div className="mb-4 flex justify-center">
      <div className="desktop:mb-[60px] mb-[26px]">
        <h2 className="text-26 desktop:mb-[8px] desktop:text-36 mb-[10px] font-bold leading-[1.2] tracking-[-0.02em]">
          {text} {/* Użyj prop 'text' */}
        </h2>
      </div>
    </div>
  );
};
```

And then in ClickOnMe.tsx you add along with the import.

`<ClickOnMeTitle text="Kliknij na mnie" />`


Changing components to type in TypeScript has several important benefits that can increase the quality and stability of your code. Here are some reasons why you should do so:

Typing:

Defining types for props allows you to better understand what data should be passed to the component. It also makes troubleshooting easier, as errors related to types are caught during compilation rather than at runtime.
Documentation:

Types act as a form of documentation. Other developers (or you yourself in the future) can more easily understand what data is required by the component, making the code more readable.

Refactoring:

When components are well typed, refactoring is easier. Changes to data types automatically inform you of places in your code that need to be updated.
Error avoidance:

Types can help catch potential errors, such as passing the wrong data types to components, which can lead to execution errors.

Reusability:

Typed components are more flexible and can be more easily used in different contexts because they are clear in their requirements for props.

PiotrDrechsler reviewed (changes requested):
Well done @anetagloc ! You have nicely divided the section into smaller components, the appearance of windows with details when you press a button works very nicely ;) I only have a few comments:

1.Can you still transfer the descriptions from the active boxes to the board in a separate component? This would significantly improve the readability of the code.An example board could look like this:

```
const services: Service[] = [
  {
    id: 1,
    title: "Usuwanie kamienia",
    description: "Oferujemy usługę usuwania kamienia nazębnego nowoczesną metodą Emmy PET, niewymagającej narkozy.",
    position: { left: "250px", top: "20px" },
  },
  {
    id: 2,
    title: "Czyszczenie uszu",
    description: "To proces usuwania nagromadzonego wosku, brudu i zanieczyszczeń z przewodów słuchowych i małżowin usznych, a także na usuwaniu włosów wewnątrz ucha.",
    position: { left: "207px", top: "60px" },
  },
  {
    id: 3,
    title: "Kąpiel",
    description: "Oczyszczanie sierści i skóry z brudu, zanieczyszczeń, pasożytów i martwego naskórka, co pomaga utrzymać ich zdrowie, komfort i przyjemny zapach.",
    position: { left: "350px", top: "280px" },
  },
  {
    id: 4,
    title: "Usługi fryzjerskie",
    description: "Czesanie, rozczesywanie, strzyżenie i trymowanie to nasze zabiegi podstawowe, wykonywane w ramach dbania o zdrowie, komfort i dobre samopoczucie Twojego czworonoga.",
    position: { left: "207.6px", top: "360px" },
  },
  {
    id: 5,
    title: "Przycięcie pazurków",
    description: "Polega na usuwaniu nadmiaru ich zrogowaciałej części, co zapobiega dyskomfortowi, bólom i problemom z poruszaniem się, a także chroni meble i podłogi przed zarysowaniami.",
    position: { left: "367px", top: "660px" },
  },
]
```

Then you import it into the main component and map it after each element of the array. Just don't forget the typing! 

```
interface Service {
  id: number;
  title: string;
  description: string;
  position: { left: string; top: string };
}
```
And an example of mapping:

```
  {services.map((service) => (
                <div
                  key={service.id}
                  className={`absolute cursor-pointer`}
                  style={{ left: service.position.left, top: service.position.top }}
                  onClick={() => toggleBox(service.id)}
                >
                  <ClickOnMeIcon />
                  {activeBox === service.id && (
                    <>
                      <div className={`absolute w-[150px] h-[2px] bg-greyLines`} style={{ top: `${parseFloat(service.position.top) - 4}px`, left: "90px" }} />
                      <div className={`absolute w-[300px] bg-cream p-4 shadow-clickonmeshadow rounded-md`} style={{ top: `${parseFloat(service.position.top) + 40}px`, left: "207px" }}>
                        <div className="flex items-center">
                          <div className="w-[2px] h-[40px] bg-yellow mr-4" />
                          <h4 className="text-20 ml-2 p-2 w-[300px]">{service.title}</h4>
                        </div>
                        <p className="text-16 p-2">{service.description}</p>
                      </div>
                    </>
                  )}
                </div>
              ))}
```
Of course you can realise your vision:) I just gave a quick example. These changes will make the code more readable, and it will be easy to introduce new sections without fear of errors (typescript watches out :D)

2.Would it be a problem for you to introduce animations for displaying the detail windows? I think it would be a nice addition to positively influence the visual perception of the site :)

3.Tailwind has cool and simple built-in effects that you can use for this purpose. But if you have your own tried and tested library, go ahead and use it :)

4.You can still create a separate component with an active box. Then it will already be super simplified and the component will look perfect :D 

If you have any questions about my comments then write to me at signal :) If you have any more ideas is to simplify the code then go ahead! I see some more possibilities here :)

Delicadency reviewed (changes requested):
A solidly written section :) 

I've added a few comments from myself regarding design representation and accessibility. Also note the components with svg paths, two of them are getting errors in the console.

Also, as @PiotrDrechsler  wrote, animations for popovers would be useful =)

anetagloc commented:
I added a new ClickOnMeActiveBox component, I added animation to the tiles. I implemented all the fixes that were added in the code review. I improved the code a little bit in terms of accessibility and screen readers.

PiotrDrechsler reviewed (approved):
Improvements made perfectly @anetagloc ! :) Great that you added aria-labels. This will help when working on the accessibility of the site. The animations really do the job. You can merge :)

Delicadency reviewed (approved):
All corrections implemented perfectly. Good job, you have the green light from me :)
